frontend/cli/cmd_download.go

```diff
@@ -2,17 +2,24 @@ package main
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/TBD54566975/ftl/backend/controller/artefacts"
 	"github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1/ftlv1connect"
 	"github.com/TBD54566975/ftl/internal/download"
 	"github.com/TBD54566975/ftl/internal/model"
 )
 
 type downloadCmd struct {
-	Dest       string              `short:"d" help:"Destination directory." default:"."`
-	Deployment model.DeploymentKey `help:"Deployment to download." arg:""`
+	Dest       string                   `short:"d" help:"Destination directory." default:"."`
+	Deployment model.DeploymentKey      `help:"Deployment to download." arg:""`
+	Registry   artefacts.RegistryConfig `embed:""`
```

Comment:
Needs a prefix.

I feel like we should be injecting the registry instance itself rather than the config? ie. put the config in the root CLI struct, and add a bind function for this.

Reply:
This and release will be the only commands that directly interact with the registry though, in most other cases we would construct it but not use it.

Reply:
It won't construct it unless it's injected

Reply:
Even so won't it 'pollute' all the other commands with OCI flags that do nothing?

Reply:
In that regard it's similar to the controller endpoint, which is unused by some commands but used in enough places that it warrants being global. This will be used in at least 5 locations that I'm aware of - download, serve/dev, release, deploy, box

Reply:
serve/dev don't use it, as they start their own registry, deploy uploads through the controller so that the local user does not need OCI credentials. Thinking about it I am not sure if we want download / release to actually interact directly with the registry long term either, rather than going through a controller so the controller can gate the interactions.

Reply:
I kind of agree, in which case we should keep the current gRPC interface...

Reply:
And continue to use it.

Reply:
(sorry, for downloads - WDYT?)